### ISSUE TITLE
apt_repository state=present doesn't create source.list file

### ISSUE THREAD

**Issue by techtonik** [ansible/ansible#6110]
##### Summary:

If there are no entries left in `source.list` after `state=absent` processing, `apt_repository` command removes the file, which makes is necessary for subsequent tasks (i.e. with `state=present`) to recreate it.
##### Steps To Reproduce:

```

---
- hosts: localhost
  tasks:
  - name: set /etc/apt/sources.list
    copy: dest=/etc/apt/sources.list content="deb http://ftp.mgts.by/pub/debian/ squeeze main non-free contrib"

  - name: remove old apt repositories
    apt_repository: repo="deb http://ftp.mgts.by/pub/debian/ squeeze main non-free contrib" state=absent

  - name: add new apt sources
    apt_repository: repo="deb http://ftp.mgts.by/pub/debian/ wheezy main contrib" state=present
```
##### Actual Results:

```
failed: [localhost] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-1392905603.55-39122886027061/apt_repository", line 1411, in <module>
    main()
  File "/root/.ansible/tmp/ansible-1392905603.55-39122886027061/apt_repository", line 356, in main
    sourceslist = SourcesList()
  File "/root/.ansible/tmp/ansible-1392905603.55-39122886027061/apt_repository", line 115, in __init__
    self.load(self.default_file)
  File "/root/.ansible/tmp/ansible-1392905603.55-39122886027061/apt_repository", line 179, in load
    with open(file, 'r') as f:
IOError: [Errno 2] No such file or directory: '/etc/apt/sources.list'


FATAL: all hosts have already failed -- aborting
```

**Comment by resmo** [comment]
seems already been fixed in devel 763c0305c170e2bb091b17c9277a80013aaaed6e

**Comment by techtonik** [comment]
Cool. When in the plan is to release it to public?

**Comment by resmo** [comment]
I am not from ansible ltd, but afaik the release is in march. But it would be great if we get another minor release 1.4.x with this fix in it. You could eather use devel in the meantime or cherry-pick the change.

**Comment by techtonik** [comment]
Yes. I thought about the same. What is the community process of backporting fixes to stable branch on GitHub?

**Comment by techtonik** [comment]
I confirm this is fixed in 1.5.0.
